lib/fetch/request.js

```diff
@@ -406,26 +406,26 @@ class Request {
     if (Object.keys(init).length !== 0) {
       // 1. Let headers be a copy of this’s headers and its associated header
       // list.
-      let headers = new Headers(this.headers)
+      let headers = new Headers(this[kHeaders])
 
       // 2. If init["headers"] exists, then set headers to init["headers"].
       if (init.headers !== undefined) {
         headers = init.headers
       }
 
       // 3. Empty this’s headers’s header list.
-      this[kState].headersList = new HeadersList()
-      this[kHeaders][kHeadersList] = this[kState].headersList
+      this[kHeaders] = new Headers()
+      this[kState].headersList = this[kHeaders][kHeadersList]
 
       // 4. If headers is a Headers object, then for each header in its header
       // list, append header’s name/header’s value to this’s headers.
       if (headers.constructor.name === 'Headers') {
-        for (const [key, val] of headers[kHeadersList] || headers) {
+        for (const [key, val] of headers) {
```

Comment:
header's Symbol.iterator is a reference to a map's Symbol.iterator, seemed like a weird hack that was added last time this was an issue (since `headers[kHeadersList]` will always be set cuz it's a Headers instance).